### PR TITLE
:sparkles:  Create EntityManager and simples entities for the server game logic (EPI-86/EPI-87/EPI-88)

### PR DESCRIPTION
**Pull request by Marius-P1**
This pull request includes significant changes to the `R-Type-Reborn` project, focusing on restructuring the build system, adding new client and server components, and defining entity classes for game logic. The most important changes include updates to the CMake configuration, the addition of client and server build targets, and the implementation of various entity classes.

### Build System Updates:
* [`.editorconfig`](diffhunk://#diff-0947e2727d6bad8cd0ac4122f5314bb5b04e337393075bc4b5ef143b17fcbd5bL4-R11): Added specific indentation rules for `CMakeLists.txt` and C++ files.
* [`CMakeLists.txt`](diffhunk://#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR6-R20): Updated to require C++17, added library path configuration, and restructured the project to include client and server subdirectories.

### Client and Server Components:
* [`client/CMakeLists.txt`](diffhunk://#diff-8404a76cc315289a505a0b9e571242cb4f5ba0c122fb86382329a4b714425ac3R1-R9): Added configuration to build the client executable and link it with the `StellarForge` library.
* [`server/CMakeLists.txt`](diffhunk://#diff-557b764d93c625467726335e930bdb5e8b625a3709e76d8afbcac860d455e5c8R1-R26): Added configuration to build the server executable and link it with the `StellarForge` library.
* [`client/main.cpp`](diffhunk://#diff-0bd51ef4d3e948a95ca277af0747d450e561ef0a1c781760b2b9e440bb9e9f88L9-R13): Updated to include a simple `Hello, World!` message and identify as the client.
* `server/Server.cpp` and `server/Server.hpp`: Added initial server logic placeholders. [[1]](diffhunk://#diff-60e56a63efe75fd8642b1fe899a068172365e68573ceb198acae2f1e1f79be45R1-R8) [[2]](diffhunk://#diff-746fc85f0e6bebf95d44c1f166e17d6b1b011e79e9126f5cc62d8503429d1093R1-R17)

### Entity Classes:
* `server/entities/ABullet.cpp` and `server/entities/ABullet.hpp`: Implemented the `ABullet` class with methods for handling bullet damage and movement. [[1]](diffhunk://#diff-070920dc8ca386ccc6501d682e50b3249b61852bce381bfbb2c32dee698c3809R1-R28) [[2]](diffhunk://#diff-c59bd5f3c2b05f8c02628c5e15b32577d039c7eb411efd915588cc3354ab2e33R1-R32)
* `server/entities/AEntity.cpp` and `server/entities/AEntity.hpp`: Implemented the `AEntity` base class with properties and methods for game entities. [[1]](diffhunk://#diff-11484729074b2468098e5e50faa4b63d62da04580195a85146bfd7deb72f9068R1-R120) [[2]](diffhunk://#diff-8eac8522be9ac7fa39f39e64f6204bd3baf21af7251cd27d080e3bc2f8204bcfR1-R49)
* `server/entities/AShip.cpp` and `server/entities/AShip.hpp`: Implemented the `AShip` class with properties and methods specific to ship entities. [[1]](diffhunk://#diff-6bbb6de2fbc320714bc1b7869b9384b3a64a5a8f529b56bd976275f54faa3164R1-R63) [[2]](diffhunk://#diff-94973c4a471bd2004d34d7a410335187a1b1c8608f684eb19792f31e53fc58acR1-R39)
* `server/entities/BasicEnemy.cpp` and `server/entities/BasicEnemy.hpp`: Implemented the `BasicEnemy` class with shooting behavior. [[1]](diffhunk://#diff-247c1fda593c709e3b3830e091bd78013f7af5b7ff4e74494a232e26812ac432R1-R36) [[2]](diffhunk://#diff-14c83950c218c0c0c8c69ed6dc05103cc6fef754c812a5985c33afce4297b14cR1-R25)
* `server/entities/ClassicBullet.cpp` and `server/entities/ClassicBullet.hpp`: Implemented the `ClassicBullet` class as a specific type of bullet. [[1]](diffhunk://#diff-0d99686a6b822192130c2c4a28a23f64a1341a5513a1a78893f82e7b20229428R1-R19) [[2]](diffhunk://#diff-f72a0c16e89a93cc6a1cada851fbf5cb3ec43f1b42887ed42df9be91a57fbc58R1-R22)
* `server/entities/EntityManager.cpp` and `server/entities/EntityManager.hpp`: Implemented the `EntityManager` class to manage game entities. [[1]](diffhunk://#diff-ea0a0b70dfac690220d1a16231d26b2d26cc101b35d943b546bd80ff591b6625R1-R55) [[2]](diffhunk://#diff-ae3f1228fee0f08c84465f76ed4e54b88aff5f5773ecc99daa8d6e8f9fb2f96dR1-R38)
* [`server/entities/IEntity.hpp`](diffhunk://#diff-ec8539a0713a4a82a4e0c832e2411d8a272ef801b1a42a6c6d770077096dde01R1-R42): Defined the `IEntity` interface for game entities.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -3,11 +3,18 @@ cmake_minimum_required(VERSION 3.5)
...
         LANGUAGES CXX)
 set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 include_directories(includes)
-link_directories(libs)
-link_libraries(StellarForge)
 
-add_executable(R-Type-Reborn ${R-Type-Reborn_SOURCES})
+set(LIBRARY_PATH "${CMAKE_CURRENT_SOURCE_DIR}/libs")
```
Comment (axel-eck) [inline CMakeLists.txt:10]: link_directories does what you want no need of variables

**Review by landryarki — APPROVED** [review]
a good temporary thing to have a working game but it will have to be replaced later on :+1:

**Review by 6im0n — APPROVED** [review]
Good for me !

File: CMakeLists.txt (hunk 1 above)
Comment (Marius-P1) [reply CMakeLists.txt:10]: This part will be update later when we are going to link the server/client part with our Game engine but for the moment it's not used

**Pull request merged**